### DISCUSSION TITLE
Intercepting Routes fail silently when (empty) `pages` directory is still present

### DISCUSSION THREAD

**Discussion by marcotoniut** [vercel/next.js#65569] [Help]
### Link to the code that reproduces this issue

https://github.com/478732/temp

### To Reproduce

I wasn't able to create a definitive minimal example; the linked repository reflects my personal project attempts. Despite forking it, adding an empty `pages` directory, and relocating the app under src, the setup still didn't replicate the issue accurately.

1. Migrate Old Pages: Start with a Next.js project that used the traditional `pages` directory. Delete everything inside this directory and switch your project to use the new `app` directory structure.
2. Attempt Intercepting Routes: Implement modal intercept functionality. Despite various attempts and configurations, the interception doesn't work as expected.
3. Remove pages Directory: Eventually, delete the empty `pages` directory entirely. After this change, the route interception starts functioning correctly.

I'm not exactly sure what the exact terms you need to comply in order to reproduce it, but I thought it might be worthwhile raising it, as some other people have seen bizarre effects when it comes to intercepting routes.

### Current vs. Expected behavior

Intercept modal route not working under some very specific circumstances

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:31 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T8112
  Available memory (MB): 8192
  Available CPU cores: 8
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 4.1.1
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Parallel & Intercepting Routes

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by ztanner** [comment]
Hi @marcotoniut -- I haven't been able to reproduce this. Do you still have an environment where it's not working for you? If so, have you tried clearing out `.next/cache`? I'm curious if some local development cache got into a bad place. This can happen when moving/deleting folders.

I'm going to move this to a discussion for now until we can determine if this is reproducible somehow.

**Comment by marcotoniut** [comment]
@ztanner , thank you for moving it to the correct place.

I tried my best to get a clean minimalistic repo to produce the same result, but couldn't do it, sadly. When attempting to solve the issue on my work repo, I did delete `.next/cache` at one point, but that didn't fix it. Sorry I couldn't provide any more details about this; I wanted to document it because of how frustrating it was to deal with, and someone else might encounter a similar problem.
